src/hotspot/share/opto/memnode.cpp

```diff
@@ -2787,22 +2787,30 @@ class MergePrimitiveStores : public StackObj {
 private:
   PhaseGVN* const _phase;
   StoreNode* const _store;
+  // State machine with initial state Unknown
+  // Allowed transitions:
+  //   Unknown  -> Forward
+  //   Unknown  -> Backward
+  //   Forward  -> Forward
+  //   Backward -> Backward
+  enum ValueOrder : uint8_t { Unknown, Forward, Backward };
```

Comment:
I think you want to express here if the parts of a value are stored in platform byte order or in reverse platform byte order. These orders are defined. Wheras `Forward` and `Backward` order are not directly clear to me.
I'd suggest to rename as follows or explain `Forward` and `Backward`
```suggestion
  enum ValueOrder : uint8_t { Unknown, Platform, Reverse };
```